### PR TITLE
87 rpi server refactor

### LINKED ISSUE #87: Review RPi running as server/client
# Goal

Review if it's possible to make the RPi the notification server as this would allow multiple devices running our client to receive notifications.

### LINKED ISSUE #116: Pedantic compiler: remove semi-colon after test-case function call
We don't need the `;` at the end of a test case function.

# Additional information

```
PosturePerfection/test/test_posture_estimator.cpp:54:2: warning: extra ‘;’ [-Wpedantic]
   54 | };
      |  ^
PosturePerfection/test/test_posture_estimator.cpp:76:2: warning: extra ‘;’ [-Wpedantic]
   76 | };
      |  ^
PosturePerfection/test/test_posture_estimator.cpp:92:2: warning: extra ‘;’ [-Wpedantic]
   92 | };
```

### PR DESCRIPTION
**Pull request by C-Begley**
# Details

- Raspberry Pi will now use Notify Broadcast to send messages to any device listening with Notify Receiver

# Checklist

## Code
- [ ] I have added some tests
- [x] I have run the tests locally to ensure they all pass by running `.scripts/build.sh -enable-testing`

## Documentation
- [x] I have updated documentation appropriately (GitHub Pages)
- [x] I have tested the updated GitHub Pages locally to ensure the changes to documentation render correctly (see this [Guidance](https://ese-peasy.github.io/PosturePerfection/guidance.html) page)

Closes #87 
Closes #116

### CODE REVIEW

File: src/notifications/broadcast_example.cpp [hunk 1]
```diff
@@ -19,12 +19,12 @@
...
+#include "broadcast.h"
 
 int main(int argc, char *argv[]) {
-  char *ip = const_cast<char *>("127.0.0.1");
-  Notify::NotifyClient client(ip, 8080);
-  std::string msg = "Hello World";
-  client.sendMessage(msg);
+  char *ip = const_cast<char *>("192.168.1.255");
+  Notify::NotifyBroadcast broadcaster(ip, 121121);
+  std::string msg = "Your back is broken :-( \n April Fools!";
```
Comment (miklasr) [inline src/notifications/broadcast_example.cpp]: erm

**Review by miklasr — CHANGES_REQUESTED** [review]
I'll try it out now, but I have some initial feedback.

**Comment by miklasr** [comment]
When running `./scripts/build.sh`
```
CMake Error at src/CMakeLists.txt:46 (add_library):
  Cannot find source file:

    notifications/client.cpp
```

**Comment by C-Begley** [comment]
> When running `./scripts/build.sh`
> 
> ```
> CMake Error at src/CMakeLists.txt:46 (add_library):
>   Cannot find source file:
> 
>     notifications/client.cpp
> ```

I'll fix that, but to test for now run

```
cd src/notifications/
cmake build .
make
```

Run `NotificationReceiver` in root posture perfection directory
Run `NotificationBroadcaster` anywhere

File: src/notifications/broadcaster.h [hunk 2]
```diff
@@ -20,38 +19,40 @@
...
+ * `Notify::NotifyReceiver`
  *
  */
-class NotifyClient {
+class NotifyBroadcast {
  private:
-  int client_fd;               ///< Socket which the client is bound to
-  struct sockaddr_in address;  ///< Client address structure
-  char* server_ip;             ///< IP address of server to connect client too
+  int broadcast_fd;            // Socket which the broadcaster is bound to
```
Comment (Ashwin-MJ) [inline src/notifications/broadcaster.h]: I think these inline comments might need to be `///<` for Doxygen to pick them up?

File: src/notifications/receiver.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    std::string cwd(current_d);
+    std::string start("notify-send \"Posture Perfection\" \"");
+    std::string middle("\" -u critical --icon ");
+    std::string end(
+        "/docs/images/"
+        "posture-logo-no-text.png");
+    this->buffer[read_value] = '\0';
+    std::string out = start + this->buffer + middle + cwd + end;
+    system(out.c_str());
+    std::cout << out;
```
Comment (Ashwin-MJ) [inline src/notifications/receiver.cpp]: Do we need it to print the command to the terminal?

File: src/notifications/broadcast_example.cpp (hunk 1 above)
Comment (C-Begley) [reply src/notifications/broadcast_example.cpp]: Just celebrating the day. Changed back now

File: src/notifications/receiver.cpp (hunk 3 above)
Comment (C-Begley) [reply src/notifications/receiver.cpp]: No we don't anymore

**Review by miklasr — APPROVED** [review]
Tested with RPi as sender and laptop as receiver and it worked very nicely.

**Review by Ashwin-MJ — APPROVED** [review]

**Review by Andrew-Ritchie — APPROVED** [review]
Looks great, good job! :)

**Pull request merged**